File: BACKEND/src/main/java/com/ROOMIFY/Roomify/model/Property.java
package com.ROOMIFY.Roomify.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "properties")
@Getter
@Setter
public class Property {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private String address;
    private double latitude;
    private double longitude;

    @Column(name = "owner_id")
    private Long ownerId;

    @Column(name = "owner_name")
    private String ownerName;

    @Column(name = "contact_phone")
    private String contactPhone;

    @Column(name = "contact_email")
    private String contactEmail;

    @Column(name = "property_type")
    private String propertyType; // BUILDING, SINGLE_ROOM, HOUSE, etc.

    @Column(name = "has_video")
    private boolean hasVideo;

    @Column(name = "has_contract")
    private boolean hasContract;

    @Column(name = "video_url")
    private String videoUrl;

    @Column(name = "contract_url")
    private String contractUrl;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @OneToMany(mappedBy = "property", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Room> units = new ArrayList<>();

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "property_images", joinColumns = @JoinColumn(name = "property_id"))
    @Column(name = "image")
    private List<String> images = new ArrayList<>();

    public Property() {}
}

File: FRONTEND/shared/src/commonMain/kotlin/org/com/model/Property.kt
package org.com.model

import kotlinx.serialization.Serializable

@Serializable
data class Property(
    val id: Long? = null,
    val title: String? = null,
    val description: String? = null,
    val address: String? = null,
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val ownerId: Long? = null,
    val ownerName: String? = null,
    val contactPhone: String? = null,
    val contactEmail: String? = null,
    val propertyType: String = "SINGLE_ROOM",
    val hasVideo: Boolean = false,
    val hasContract: Boolean = false,
    val videoUrl: String? = null,
    val contractUrl: String? = null,
    val images: List<String> = emptyList(),
    val units: List<Room> = emptyList()
)
